ui/guest/src/store/reducers/root.ts

```diff
@@ -110,6 +111,14 @@ const resetState: (state: GuestState) => GuestState = (state: GuestState) => ({
   dragContext: null
 });
 
+const filterDropZones = (dropZones: DropZone[], instance) => {
+  return dropZones.filter((dropZone) => {
+    const instanceId = instance.craftercms.id;
+    const dropZoneId = iceRegistry.getById(dropZone.iceId).modelId;
+    return !contentController.modelHierarchyMap[dropZoneId].children.includes(instanceId);
+  });
+};
```

Comment:
We already have some reducers which aren't pure but we don't want to keep adding to them. These should be done earlier, before dispatching the action; whether it is at the epic or handler level.